### PR TITLE
Add Regression Test for Reported Undo/Redo Regression

### PR DESCRIPTION
**Pull request by anthony-murphy**
There was a reported regression in the existing/old undo redo stack around deleted segments, which are known to have problems. Added a regression to the new revertible infa which should resolve the problems around deleted segments. The issue does involve marker segments which should not matter, as both the old and new infra is agnostic to marker types. This was confirmed by the regression tests

[AB#2720](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2720)

### CODE REVIEW

File: packages/dds/merge-tree/src/test/testClientLogger.ts [hunk 1]
```diff
@@ -264,31 +264,37 @@ export class TestClientLogger {
...
+                            acked += " ";
+                            local += " ";
                         }
+                        parent = node.parent;
+                    }
+                    const text = 
+                        TextSegment.is(node) 
+                            ? node.text 
+                            : Marker.is(node) 
+                                ? "¶"
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/test/testClientLogger.ts:278]: this just adds output support for marker nodes. we represent them with the paragraph symbol `¶`

File: packages/dds/merge-tree/src/test/revertibles.spec.ts [hunk 2]
```diff
@@ -235,4 +239,107 @@ describe("MergeTree.Revertibles", () => {
 
         logger.validate();
     });
+
+    describe("Redo does not works as expected when a pair of markers and text is involved", ()=>{
+
+        generatePairwiseOptions({
+            revertMarkerInsert:[true, undefined],
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/test/revertibles.spec.ts:246]: it wasn't 100% clear from the report, which operations we're reverted, and if they were local or ack. this pairwise matrix gives us good coverage across the possible combinations

File: packages/dds/merge-tree/src/test/revertibles.spec.ts [hunk 3]
```diff
@@ -235,4 +239,107 @@ describe("MergeTree.Revertibles", () => {
 
         logger.validate();
     });
+
+    describe("Redo does not works as expected when a pair of markers and text is involved", ()=>{
```
Comment (Abe27342) [inline packages/dds/merge-tree/src/test/revertibles.spec.ts:243]: seems negated from what it should be...?

```suggestion
    describe("Redo works as expected when a pair of markers and text is involved", ()=>{
```

File: packages/dds/merge-tree/src/test/revertibles.spec.ts [hunk 4]
```diff
@@ -235,4 +239,107 @@ describe("MergeTree.Revertibles", () => {
...
+                const old = clients.B.mergeTreeDeltaCallback;
+                const clientBDriver = createRevertDriver(clients.B);
+                clientBDriver.submitOpCallback = (op) => ops.push(clients.B.makeOpMessage(op, ++seq));
+                clients.B.mergeTreeDeltaCallback = (op, delta) => {
+                    old?.(op, delta);
+                    if (op.sequencedMessage === undefined && clientB_Revertibles.length > 0) {
+                         appendToMergeTreeDeltaRevertibles(
+                            clientBDriver, 
+                            delta, 
+                            clientB_Revertibles[0]);
```
Comment (Abe27342) [inline packages/dds/merge-tree/src/test/revertibles.spec.ts:274]: I missed that you were grabbing the 0th entry of clientB_Revertibles on first read. took me a minute to realize the calls to `unshift` are to set up a new undo-redo transaction. I think the intent is a bit more obvious if you define a `const openNewUndoRedoTransaction = () => clientB_Revertibles.unshift([]);` and use that instead wherever you need.

**Review by Abe27342 — APPROVED** [review]
couple nits, looks good otherwise.

**Pull request merged**